Modules/signalmodule.c

```diff
@@ -1607,9 +1607,39 @@ PyErr_CheckSignals(void)
 }
 
 
-/* Replacements for intrcheck.c functionality
- * Declared in pyerrors.h
- */
+/* Simulate the effect of a SIGINT signal arriving. The next time
+   PyErr_CheckSignals() is called,  the Python SIGINT signal handler will be
+   raised.
+
+   The SIGINT signal must be handled by Python, otherwise an exception is
+   raised and return -1. Return 0 on success.
+
+   The GIL doesn't need to be hold to call this function. */
```

Comment:
You must hold the GIL to call PyErr_SetString().

Reply:
@vstinner That function seems to be called only in [thread_PyThread_interrupt_main](https://github.com/mcepl/cpython/blob/7e414f2f495e73258f55bdef153bc19fb8a9e472/Modules/_threadmodule.c#L1106) and I am really not sure what's the state of GIL there. Of course, the line in the comment should go.

Reply:
Right. The comment is gone.